src/toil/batchSystems/torque.py

```diff
@@ -0,0 +1,189 @@
+
+        def getJobExitCode(self, torqueJobID):
+            job, task = torqueJobID
+            args = ["qstat", "-f", str(job)]
+
+            process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            for line in process.stdout:
+                line = line.strip()
+                if line.startswith("failed") and int(line.split()[1]) == 1:
+                    return 1
+                if line.startswith("exit_status"):
+                    status = line.split(' = ')[1]
+                    logger.debug('Exit Status: ' + status)
+                    return int(status)
+            return None
+
+        '''
+        Implementation-specific helper methods
+        '''
+        def prepareQsub(self, cpu, mem, jobID):
+
+            # TODO: passing $PWD on command line not working for -d, resorting to
+            # $PBS_O_WORKDIR but maybe should fix this here instead of in script?
+
+            # TODO: we previosuly trashed the stderr/stdout, as in the commented
+            # code, but these may be retained by others, particularly for debugging.
+            # Maybe an option or attribute w/ a location for storing the logs?
+
+            # qsubline = ['qsub', '-V', '-j', 'oe', '-o', '/dev/null',
+            #             '-e', '/dev/null', '-N', 'toil_job_{}'.format(jobID)]
+
+            qsubline = ['qsub', '-V', '-N', 'toil_job_{}'.format(jobID)]
+
+            if self.boss.environment:
+                qsubline.append('-v')
+                qsubline.append(','.join(k + '=' + quote(os.environ[k] if v is None else v)
+                                         for k, v in self.boss.environment.iteritems()))
+
+            reqline = list()
+            if mem is not None:
+                memStr = str(mem / 1024) + 'K'
+                reqline += ['-l mem=' + memStr]
+
+            if cpu is not None and math.ceil(cpu) > 1:
+                qsubline.extend(['-l ncpus=' + str(int(math.ceil(cpu)))])
+
+            return qsubline
+
+        def qsub(self, qsubline):
+            process = subprocess.Popen(qsubline, stdout=subprocess.PIPE)
+            so, se = process.communicate()
+            # TODO: the full URI here may be needed on complex setups, stripping
+            # down to integer job ID only may be bad long-term
+            result = int(so.strip().split('.')[0])
+            return result
+
+        def _generateTorqueWrapper(self, command):
+            """
+            A very simple script generator that just wraps the command given; for
+            now this goes to default tempdir
+            """
+            _, tmpFile = tempfile.mkstemp(suffix='.sh', prefix='torque_wrapper')
+            fh = open(tmpFile , 'w')
+            fh.write("$!/bin/bash\n\n")
+            fh.write("cd $PBS_O_WORKDIR\n\n")
+            fh.write(command + "\n")
+            fh.close
+            return tmpFile
+
+    """
+    The interface for the PBS/Torque batch system
+    """
+
+    @staticmethod
```

Comment:
We discourage use of @staticmethod and prefer @classmethod. It's incidental that a method doesn't access  class attributes and therefore has no use for the `cls` parameter. And because its incidental, always using @classmethod avoids unnecessary refactorings in the future when changes to that method do suddenly require the `cls` argument.